process.py

```diff
@@ -125,13 +126,22 @@ def system_profile(hostname, cpu_info, virt_what, meminfo, ip_addr, dmidecode,
         for yum_repo_file in yum_repos_d:
             for yum_repo_definition in yum_repo_file:
                 baseurl = yum_repo_file[yum_repo_definition].get('baseurl', [])
-                repo = {'name': yum_repo_file[yum_repo_definition].get('name'),
+                repo = {'id': yum_repo_definition,
+                        'name': yum_repo_file[yum_repo_definition].get('name'),
                         'base_url': baseurl[0] if len(baseurl) > 0 else None,
                         'enabled': _to_bool(yum_repo_file[yum_repo_definition].get('enabled')),
                         'gpgcheck': _to_bool(yum_repo_file[yum_repo_definition].get('gpgcheck'))}
                 repos.append(_remove_empties(repo))
         profile['yum_repos'] = repos
 
+    if dnf_modules:
+        modules = []
```

Comment:
should this be an empty list? The for loop below will not have anything to iterate over right?

Reply:
It should iterate over dnf_modules. fixed